pkg/reconciler/taskrun/taskrun_test.go

```diff
@@ -272,6 +273,166 @@ func getTaskRunController(t *testing.T, d test.Data) (test.TestAssets, func()) {
 	}, cancel
 }
 
+func TestReconcile_ExplicitDefaultSA(t *testing.T) {
+	taskRunSuccess := tb.TaskRun("test-taskrun-run-success", "foo", tb.TaskRunSpec(
+		tb.TaskRunTaskRef(simpleTask.Name, tb.TaskRefAPIVersion("a1")),
+	))
+	taskRunWithSaSuccess := tb.TaskRun("test-taskrun-with-sa-run-success", "foo", tb.TaskRunSpec(
+		tb.TaskRunTaskRef(saTask.Name, tb.TaskRefAPIVersion("a1")),
+		tb.TaskRunServiceAccount("test-sa"),
+	))
+	taskruns := []*v1alpha1.TaskRun{taskRunSuccess, taskRunWithSaSuccess}
+	d := test.Data{
+		TaskRuns: taskruns,
+		Tasks:    []*v1alpha1.Task{simpleTask, saTask},
+	}
+
+	defaultSAName := "pipelines"
+	defaultCfg := &config.Config{
```

Comment:
@bobcatfish @vdemeester Does this suffice?  Has 2 tests
  1. verifies that default service account specified in defaults is used
  1. verifies that  if  SA is set ( `test-sa` ) then that will be used